2-mandatory/2-water-bottle.js

```diff
@@ -17,20 +17,43 @@ You have to implement the missing features according to the specification.
 // Here is your starting point:
 let bottle = {
   volume: 0,
+
   fillUp: function () {
-    // calling this function should completely fill your bottle (volume = 100);
+     // calling this function should completely fill your bottle (volume = 100);
+     this.volume = 100;
   },
+
   pour: function () {
-    // calling this function should increase your bottle volume by 10 units;
+     // calling this function should increase your bottle volume by 10 units;
+     if (this.volume <= 90) {
+        this.volume += 10;
+     } else if (this.volume > 90 && this.volume < 100) {
+        this.volume = 100;
+     } else {
+        this.volume = this.volume;
+     }
   },
+
   drink: function () {
-    // calling this function should decrease your bottle volume by 10 units;
+     // calling this function should decrease your bottle volume by 10 units;
+     this.volume = this.volume;
+     if (this.volume > 10) {
+        this.volume -= 10;
+     } else if (this.volume < 10 && this.volume > 0) {
+        this.volume = 0;
+     } else {
+        this.volume = this.volume;
+     }
   },
+
   isFull: function () {
-    // this function should return true if your bottle is full;
+     // this function should return true if your bottle is full;
+     return this.volume === 100 ? true : false;
```

Comment:
Do you think we need to use the ternary operator to return true/false?

this.volume === 100 already returns a binary value. So, if the volume was 90, then you have actually written:

return false ? true : false